### PR TITLE
Add test

### PR DESCRIPTION
**Pull request by cothan**
[//]: # (SPDX-License-Identifier: CC-BY-4.0)
[//]: # (TODO Customize PR template)

[//]: # (See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository )

This PR add original test from Kyber: 

1. Test API: A simple API test to make sure the API is correct.
2. Test vector: Output will be hashed and compare with KAT (to be add to Github action later). 

Future: Collect more tests (test should be under Apache 2.0 License). 

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)

### CODE REVIEW

File: test/test_vector.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/* Deterministic randombytes by Daniel J. Bernstein */
```
Comment (mkannwischer) [inline test/test_vector.c:1]: We'll need to figure out the license for this one. Like this it cannot be merged.

Comment (cothan) [reply test/test_vector.c:1]: This test is not important. I'll take it out.

Comment (cothan) [reply test/test_vector.c]: removed.

**Comment by cothan** [comment]
Done. Run `make` you will see compile binary inside `test/`.

**Comment by mkannwischer** [comment]
Yes, looking good to me. Thanks!

**Review by mkannwischer — APPROVED** [review]

**Pull request merged**